diff --git a/support.cpp b/support.cpp
index 82a2e4b..fe97177 100644
--- a/support.cpp
+++ b/support.cpp
@@ -48,9 +48,31 @@ parse_map(const char* map, const char* entities)
 {
 	FILE* fp;
 	char path[256];
+	bool has_map_filter = false;
 
 	g_Stripper.SetEntityList(entities);
 
+	for (const char* pPart = map; *pPart++ != '\0';)
+	{
+		pPart = strchrnul(pPart, '_');
+		stripper_game.path_format(path, sizeof(path), "%s/%s/maps/%.*s.cfg",
+			stripper_game.game_path, stripper_game.stripper_cfg_path, pPart - map, map);
+		fp = fopen(path, "rt");
+
+		if (fp)
+		{
+			fclose(fp);
+			has_map_filter = true;
+			break;
+		}
+	}
+
+	if (!has_map_filter)
+	{
+		stripper_game.log_message("Skipping filters for unconfigured map: %s", map);
+		return g_Stripper.ToString();
+	}
+
 	stripper_game.path_format(path,
 		sizeof(path),
 		"%s/%s/global_filters.cfg",
